src/shared/player/weapon.rs

```diff
@@ -0,0 +1,137 @@
+use avian2d::prelude::*;
+use bevy::{
+    color::palettes::css::WHITE,
+    prelude::*,
+    sprite::{MaterialMesh2dBundle, Mesh2dHandle},
+};
+use leafwing_input_manager::prelude::*;
+use lightyear::prelude::*;
+
+use crate::shared::input::PlayerAction;
+use crate::shared::FixedSet;
+
+pub struct BulletPlugin;
+
+impl Plugin for BulletPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Startup, setup_weapon)
+            .add_systems(Update, spawn_bullet_mesh.in_set(FixedSet::Main))
+            .add_systems(Update, move_bullets);
+    }
+}
+
+fn spawn_bullet_mesh(
+    time: Res<Time>,
+    mut firing_timer: Local<f32>, // Local variable to keep track of firing time
```

Comment:
Use a component instead, because there are more than just 1 player. Each player will need to use their own timer.